app/src/main/java/de/rampro/activitydiary/helpers/ActivityHelper.java

```diff
@@ -196,23 +195,24 @@ public void scheduleRefresh() {
     }
 
     public ArrayList<DiaryActivity> sortedActivities(String query) {
+        ArrayList<DiaryActivity> activities = new ArrayList<DiaryActivity>(ActivityHelper.helper.getActivities().size());
         ArrayList<DiaryActivity> filtered = new ArrayList<DiaryActivity>(ActivityHelper.helper.getActivities().size());
-        ArrayList<Integer> filteredDist = new ArrayList<Integer>(ActivityHelper.helper.getActivities().size());
-
         for(DiaryActivity a : ActivityHelper.helper.getActivities()){
-            int dist = ActivityHelper.searchDistance(query, a.getName());
-            int pos = 0;
-            // search where to enter it
-            for(Integer i : filteredDist){
-                if(dist > i.intValue()){
-                    pos++;
-                }else{
-                    break;
-                }
+            if (a.getName().toLowerCase().contains(query.toLowerCase())) {
+                activities.add(a);
             }
+        }
 
-            filteredDist.add(pos, Integer.valueOf(dist));
-            filtered.add(pos, a);
+        int counter = 0;
+        while (activities.size() != filtered.size()) {
+            for (DiaryActivity a : activities) {
+                if (counter <= a.getName().length()) {
+                    if (a.getName().toLowerCase().indexOf(query.toLowerCase()) == counter) {
+                        filtered.add(a);
+                    }
+                }
+            }
+            counter++;
```

Comment:
Ah, now I see... You wanted to change the priority of full prefix searches (maybe I should read your comment next time BEFORE I look at the code)

Actually it doesn't seem to work properly now...if I search for "hard" because I mistyped "Gardening" and hit h instead of g, it doesn't find anything at all anymore. But I totally agree, that the weighting is not proper and that a full prefix should hit "better". Nevertheless, I propose to fix that in a separate PR and even create a ticket for it. -> ActivityHelper.searchDistance would be the place to fix it. In 292c0233975911347e313b1a8351f7417695b8be I just gave some more weight on true prefixes. Maybe you wanna check that out?